### PR TITLE
Update checkbox design

### LINKED ISSUE #235: [Chore] Update Checkbox Design
### What general chore or task would you like to see performed?
The design of our checkboxes need to be updated to meet the design spec. 

###### Example 
https://tangoe.invisionapp.com/dsm/tangoe/tangoe-ds/folder/components/5d3b2fd34b5f2e3140d29587

### Are there any specific steps you would like completed?
1. Each version of the checkbox should be represented in our component

### PR DESCRIPTION
**Pull request by jaredami**
Closes #235 

We need to update the design of our `go-checkbox` to meet the design spec, seen here: https://tangoe.invisionapp.com/dsm/tangoe/tangoe-ds/folder/components/5d3b2fd34b5f2e3140d29587

### CODE REVIEW

File: projects/go-lib/src/styles/_forms.scss [hunk 1]
```diff
@@ -106,6 +113,66 @@ $input--disabled-background: rgba($base-dark-secondary, .4);
...
+
+/* When the checkbox is disabled and checked, change the background and border color of the custom checkbox */
+.go-form__checkbox:disabled:checked ~ .go-form__custom-checkbox {
+  background-color: lighten($base-light-secondary, 13.5%);
+  border: 1px solid lighten($base-light-secondary, 13.5%);
+}
+
+/* Create and style a custom checkmark (hidden when not checked) */
+.go-form__custom-checkbox:after {
+  border: solid #FDFDFD;
```
Comment (jaredami) [inline projects/go-lib/src/styles/_forms.scss]: I believe we need to create a variable for this, unless it is equivalent to one of our calculated colors.

File: projects/go-lib/src/styles/_forms.scss [hunk 2]
```diff
@@ -106,6 +113,66 @@ $input--disabled-background: rgba($base-dark-secondary, .4);
...
+  opacity: 0;
+  position: absolute;
+  width: 0;
+}
+
+/* Create a custom checkbox */
+.go-form__custom-checkbox {
+  background-color: $base-light;
+  border: 1px solid lighten($base-light-secondary, 13.5%);
+  border-radius: 2px;
```
Comment (grahamhency) [inline projects/go-lib/src/styles/_forms.scss]: I think we can push this to 3px with our `global-radius` variable. Thoughts?

File: projects/go-lib/src/styles/_forms.scss (hunk 1 above)
Comment (grahamhency) [reply projects/go-lib/src/styles/_forms.scss]: I think this should just be `$base-light`. If you look at the DSM it says `or Shuttle` and `Shuttle` is just `#fff` which is `$base-light`.

File: projects/go-lib/src/styles/_forms.scss (hunk 2 above)
Comment (jaredami) [reply projects/go-lib/src/styles/_forms.scss]: It looks noticeably more rounded than the design spec that way, but it still looks fine in my opinion. 

So I guess it comes down to how important it is that we match the design spec as closely as possible vs how much we want to keep things consistent, e.g. by using our global variables 🤷‍♂

Comment (grahamhency) [reply projects/go-lib/src/styles/_forms.scss]: I would say that we should strive for consistency. Let's use the variable.

**Pull request merged**